packages/app/src/__tests__/pages/market.test.tsx

```diff
@@ -170,4 +133,54 @@ describe('Futures market page - smart margin', () => {
 			'1829.279274630724573866'
 		)
 	})
+
+	test('Displays error when trade exceeds max OI', async () => {
+		// Update the mock to return some different data
+		sdk.futures.getMarkets = () =>
+			Promise.resolve([{ ...SDK_MARKETS[1], marketLimitUsd: wei(100000) } as FuturesMarket])
+
+		const store = setupStore(
+			preloadedStateWithSmartMarginAccount(mockSmartMarginAccount('1000000'))
+		)
+		const { findByTestId, findByText } = render(
+			<MockProviders route="market/?accountType=cross_margin&asset=sETH" store={store}>
+				<Market />
+			</MockProviders>
+		)
+
+		const marginInput = await findByTestId('set-order-margin-susd-desktop')
+		fireEvent.change(marginInput, { target: { value: '100000' } })
+
+		const sizeInput = await findByTestId('set-order-size-amount-susd-desktop')
+		fireEvent.change(sizeInput, { target: { value: '1000000' } })
+
+		// OI limit warning displayed
+		const fillPrice = await findByText('Open interest limit exceeded')
+		expect(fillPrice).toBeTruthy()
+	})
+
+	test('Trade panel is disabled when market is closed', async () => {
+		// Update the mock to return some different data
+		sdk.futures.getMarkets = () =>
+			Promise.resolve([{ ...SDK_MARKETS[1], isSuspended: true } as FuturesMarket])
+
+		const store = setupStore(preloadedStateWithSmartMarginAccount())
+		const { findByTestId, findByText } = render(
+			<MockProviders route="market/?accountType=cross_margin&asset=sETH" store={store}>
+				<Market />
+			</MockProviders>
+		)
+
+		const marginInput = await findByTestId('set-order-margin-susd-desktop')
+		fireEvent.change(marginInput, { target: { value: '100000' } })
+
+		const sizeInput = await findByTestId('set-order-size-amount-susd-desktop')
+		fireEvent.change(sizeInput, { target: { value: '1000000' } })
+
+		const submitButton = await findByTestId('trade-panel-submit-button')
+		expect(submitButton).toBeDisabled()
```

Comment:
Great test, but it would be cool to add validity to `submitButton` check. It would be `disabled` anyway, since the values of available margin and size exceed `preloadedState` (freeMargin = 1000)

Reply:
Great point, will update the test to validate an enabled state first.

Reply:
updated to check enabled first